### ISSUE TITLE
Port twisted.internet.address to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5985]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#5985|
|Type|enhancement|
|Created|2012-09-14 17:28:19Z|
|Branch|https://github.com/twisted/twisted/tree/address-py3-5985|

`twisted.internet.address` should be ported to Python 3.

<details><summary>Searchable metadata</summary>

```
trac-id__5985 5985
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_address_py3_5985 branches/address-py3-5985
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1347643699000000 1347643699000000
changetime__1347658524000000 1347658524000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [35673]) Branching to 'address-py3-5985'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review. Started build here:
http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/address-py3-5985

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @itamarst|
|-|-|

  1. One new pydoctor error - http://buildbot.twistedmatrix.com/builders/documentation/builds/2498/steps/api-documentation/logs/new%20pydoctor%20errors - maybe a pydoctor bug.  this looks like a case mwhudson said should work just a couple days ago.  Maybe importing `FancyEqMixin` stand-alone would work around it, dunno.

Otherwise looks good.  Please merge after addressing the above.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst set status to `closed`|
|-|-|

(In [35686]) Merge address-py3-5985.

Author: itamar
Review: exarkun
Fixes: [#5985](https://github.com/twisted/twisted/issues/5985)

Port twisted.internet.address to Python 3.
